test/contrib/sinatra/tracer_test.rb

```diff
@@ -30,6 +30,10 @@ class TracerTestApp < Sinatra::Application
     get '/literal-template' do
       erb '<%= msg %>', locals: { msg: 'hello' }
     end
+
+    before do
+      response.headers['X-Request-ID'] = 'request-id'
```

Comment:
Given these are integration tests, is it better to force headers here, or to pass them in from `rack-test`'s `#get` method? This kind of change makes some assumptions about how the Sinatra API works. For test resiliency, it might be beneficial to take a more black-box approach here.